### ISSUE TITLE
TypeDoc generates broken links for `{@linkcode ...}` comments

### ISSUE THREAD

**Issue by mrousavy** [typedoc2md/typedoc-plugin-markdown#482]
Hey!
Thanks for this amazing plugin, I use TypeDoc to generate API documentation for [react-native-vision-camera](https://react-native-vision-camera.com)! ❤️ 

I am using `{@linkcode }` blocks, (see code [here](https://github.com/mrousavy/react-native-vision-camera/blob/a498144fd5786c6dba1d78730d362e0f675cbaf7/package/src/CameraProps.ts#L118)), but for some reason the output links are broken:

```
Exhaustive list of all broken links found
-> linking to CameraDeviceFormat.md#maxfps (resolved as: /docs/api/interfaces/CameraDeviceFormat.md)
```

![Screenshot 2023-10-10 at 18 15 16](https://github.com/tgreyuk/typedoc-plugin-markdown/assets/15199031/bb3687de-0b82-4835-b800-b84e859e3e2a)

Is there any way to configure that? Or am I using `{@linkcode ...}` wrong?

**Comment by mrousavy** [comment]
(The problem is the `.md` suffix, it needs to be generated without `.md`, then it's a valid link)

**Comment by tgreyuk** [comment]
So I have done a bit of digging and `{@linkcode}` is parsing correctly but there does seem to be a Docusaurus bug in resolving broken links when a link is immediately followed by a parenthesis.

broken:
`[label](page.md#anchorName) (see [label](page.md#anchorName))`

ok:
`see ([label](page.md#anchorName))`

ok:
` [label](page.md#anchorName) - (see [label](page.md#anchorName))`

**Comment by mrousavy** [comment]
Interesting! Thanks for your insights, I guess this should be reported in the Docusaurus repo then?

**Comment by tgreyuk** [comment]
>  I guess this should be reported in the Docusaurus repo then?

Yes I think so - I'll do a bit more investigation and raise a ticket.

**Comment by mrousavy** [comment]
Awesome, thank you I appreciate it! 🙏

**Comment by bo-carey** [comment]
@tgreyuk - I am also experiencing this bug. Have you made a ticket?

**Comment by tgreyuk** [comment]
Thanks @bo-carey i was just looking at this and i notice you have already raised a ticket. I am linking to it here as a reference - https://github.com/facebook/docusaurus/issues/9518 .

**Comment by bo-carey** [comment]
@tgreyuk thanks for that, Tom!

**Comment by tgreyuk** [comment]
Have tested and I can confirm that this issue has been fixed in Docusaurus 3.4. https://github.com/facebook/docusaurus/pull/10168.
